repose-aggregator/tests/functional-test-framework/src/main/groovy/org/openrepose/framework/test/ReposeValveLauncher.groovy

```diff
@@ -87,6 +89,12 @@ class ReposeValveLauncher extends ReposeLauncher {
         if (params.containsKey("waitOnJmxAfterStarting")) {
             waitOnJmxAfterStarting = params.waitOnJmxAfterStarting
         }
+        if (params.containsKey("reposeXmx")) {
+            xmx = params.reposeXmx
```

Comment:
Following the other comment, then convert these to a local variable and pass along to the start method, similar to `killOthersBeforeStarting` and `waitOnJmxAfterStarting`, so we're not changing those instance variables each call.

Reply:
If I'm always passing those variables to start though, the default value wouldn't ever be used right?

Reply:
I guess I could just check if those values are in "params" and if so do this 

`start(killOthersBeforeStarting, waitOnJmxAfterStarting, nodeId, xmx, xms)`

else do `start(killOthersBeforeStarting, waitOnJmxAfterStarting, nodeId)`

Reply:
Well, the defaults on the start method would get used any time someone bypasses this method and calls `start(false, true, "some-node")` directly (which I'm pretty sure I've seen in some places?)

Reply:
A lot of the tests right now just call `repose.start()` which call the other start method `void start(Map params)`

This one calls `start(killOthersBeforeStarting, waitOnJmxAfterStarting, nodeId)`, but if a test calls it like this: `repose.start([reposeXmx:"1536M",reposeXms:"1024M"])` then we'd need to pass those values to the other start:

`start(killOthersBeforeStarting, waitOnJmxAfterStarting, nodeId, xmx, xms)` which overrides the default values. But we'd still want `repose.start()` to not override those values, so we'd need a conditional so that when the values aren't passed, it just calls `start(killOthersBeforeStarting, waitOnJmxAfterStarting, nodeId)`. 

Or I could move the default values to the `void start(Map params)` method instead of having them come in through params.

`void start(Map params, xmx="1024M", xms="512M")`

I think I might like that more actually.

Reply:
Oh, I see. I'm fine leaving it the way it is (just nitpicking).

I think I'd do this.

```groovy
// Put these on the class
static def DEFAULT_REPOSE_XMX = "1024M"
static def DEFAULT_REPOSE_XMS = "512M"

// Update this with "get or default" logic
def start(Map params) {
    ...
    def reposeXmx = params.get('reposeXmx', DEFAULT_REPOSE_XMX)
    def reposeXms = params.get('reposeXms', DEFAULT_REPOSE_XMS)
    start(killOthersBeforeStarting, waitOnJmxAfterStarting, nodeId, reposeXmx, reposeXms)
}

// Update this to have the defaults
start(boolean killOthersBeforeStarting, boolean waitOnJmxAfterStarting, String nodeId, 
      String xmx = DEFAULT_REPOSE_XMX, String xms = DEFAULT_REPOSE_XMS)
```

The `start([:])` method is still overriding the default values in the other start method, but they at least reference all the same constants so it will be the right values.